cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/client/AmazonEc2RetryClient.java

```diff
@@ -1,52 +1,184 @@
 import com.amazonaws.services.ec2.model.ModifyInstanceAttributeRequest;
 import com.amazonaws.services.ec2.model.ModifyInstanceAttributeResult;
+import com.amazonaws.services.ec2.model.StartInstancesRequest;
+import com.amazonaws.services.ec2.model.StartInstancesResult;
+import com.amazonaws.services.ec2.model.StopInstancesRequest;
+import com.amazonaws.services.ec2.model.StopInstancesResult;
+import com.amazonaws.services.ec2.model.TerminateInstancesRequest;
+import com.amazonaws.services.ec2.model.TerminateInstancesResult;
+import com.amazonaws.services.ec2.waiters.AmazonEC2Waiters;
 import com.sequenceiq.cloudbreak.service.Retry;
 
-public class AmazonEc2RetryClient extends AmazonRetryClient {
+public class AmazonEc2RetryClient extends AmazonClient {
 
-    private final AmazonEC2Client client;
+    private final AmazonEC2 client;
 
     private final Retry retry;
 
-    public AmazonEc2RetryClient(AmazonEC2Client client, Retry retry) {
+    public AmazonEc2RetryClient(AmazonEC2 client, Retry retry) {
         this.client = client;
         this.retry = retry;
     }
 
     public CreateVolumeResult createVolume(CreateVolumeRequest request) {
-        return retry.testWith2SecDelayMax15Times(() -> mapThrottlingError(() -> client.createVolume(request)));
+        return retry.testWith2SecDelayMax15Times(() -> client.createVolume(request));
     }
 
     public DescribeSubnetsResult describeSubnets(DescribeSubnetsRequest request) {
-        return retry.testWith2SecDelayMax15Times(() -> mapThrottlingError(() -> client.describeSubnets(request)));
+        return retry.testWith2SecDelayMax15Times(() -> client.describeSubnets(request));
     }
 
     public ModifyInstanceAttributeResult modifyInstanceAttribute(ModifyInstanceAttributeRequest request) {
-        return retry.testWith2SecDelayMax15Times(() -> mapThrottlingError(() -> client.modifyInstanceAttribute(request)));
+        return retry.testWith2SecDelayMax15Times(() -> client.modifyInstanceAttribute(request));
     }
 
     public DeleteVolumeResult deleteVolume(DeleteVolumeRequest request) {
-        return retry.testWith2SecDelayMax15Times(() -> mapThrottlingError(() -> client.deleteVolume(request)));
+        return retry.testWith2SecDelayMax15Times(() -> client.deleteVolume(request));
     }
 
     public DescribeVolumesResult describeVolumes(DescribeVolumesRequest request) {
-        return retry.testWith2SecDelayMax15Times(() -> mapThrottlingError(() -> client.describeVolumes(request)));
+        return retry.testWith2SecDelayMax15Times(() -> client.describeVolumes(request));
     }
 
     public AttachVolumeResult attachVolume(AttachVolumeRequest request) {
-        return retry.testWith2SecDelayMax15Times(() -> mapThrottlingError(() -> client.attachVolume(request)));
+        return retry.testWith2SecDelayMax15Times(() -> client.attachVolume(request));
+    }
+
+    public DescribeRegionsResult describeRegions(DescribeRegionsRequest describeRegionsRequest) {
```

Comment:
these methods are not retryable. Are they should retryable or maybe create another class for these methods

Reply:
I would prefer to rename the service, removing the retry from the name as it does not really matter from the point of the using class